jdbc/src/main/java/org/openconnectors/JdbcSourceConnector.java

```diff
@@ -0,0 +1,192 @@
+
+import com.google.common.collect.Sets;
+import org.openconnectors.config.Config;
+import org.openconnectors.connect.ConnectorContext;
+import org.openconnectors.connect.PushSourceConnector;
+import org.openconnectors.data.Record;
+import org.openconnectors.source.BulkJdbcQuerier;
+import org.openconnectors.source.QuerierThread;
+import org.openconnectors.source.TableQuerier;
+import org.openconnectors.util.CachedConnectionProvider;
+import org.openconnectors.util.ConnectionProvider;
+import org.openconnectors.util.DbConnectionConfig;
+import org.openconnectors.util.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class JdbcSourceConnector implements PushSourceConnector<Record> {
+
+    private static Logger logger = LoggerFactory.getLogger(JdbcSourceConnector.class);
+
+    private Config config;
+    private List<ConnectionProvider> connectionProviders = new ArrayList<>();
+    private List<QuerierThread> workers = new ArrayList<>();
+    private Consumer<Collection<Record>> consumeFunction;
+
+    private void configureConnector(Config config) {
+        // TODO: validate config
+        this.config = config;
+    }
+
+
+    @Override
+    public void initialize(ConnectorContext ctx) {
+
+    }
+
+    @Override
+    public void open(Config config) throws Exception {
+        configureConnector(config);
+        final String dbUrl = config.getString(JdbcConfigKeys.CONNECTION_URL_CONFIG);
+        final String dbUser = config.getString(JdbcConfigKeys.CONNECTION_USER_CONFIG);
+        final String dbPassword = config.getString(JdbcConfigKeys.CONNECTION_PASSWORD_CONFIG);
+        final int dbMaxConnectionAttempts = config.getInt(JdbcConfigKeys.CONNECTION_MAX_ATTEMPT);
+        final int dbConnectionRetryDelay = config.getInt(JdbcConfigKeys.CONNECTION_RETRY_DELAY);
+        final DbConnectionConfig connectionConfig = DbConnectionConfig.Builder
+                .newBuilder()
+                .setUrl(dbUrl)
+                .setUser(dbUser)
+                .setPassword(dbPassword)
+                .setMaxConnectionAttempts(dbMaxConnectionAttempts)
+                .setConnectionRetryDelay(dbConnectionRetryDelay)
+                .build();
+        ConnectionProvider connectionProvider = getConnectionProvider(connectionConfig);
+
+        Connection connection = connectionProvider.getValidConnection();
+
+        Set<String> whiteListTables = getWhiteListTables();
+        Set<String> blackListTables = getBlackListTables();
+        if (!whiteListTables.isEmpty() && !blackListTables.isEmpty()) {
+            String errorMsg = "Only white list or black list of tables must be specified, rather than both of them.";
```

Comment:
I think a better error message would be:
`Specify either "white.list.tables" or "black.list.tables", but not both`